spec/models/rule_spec.rb

```diff
@@ -20,7 +20,7 @@
       allow(rule_effect).to receive(:prepare).and_call_original
 
       rule.process(subject.id, {})
-      expect(rule_effect).to have_received(:prepare).with(workflow.id, subject.id).once
+      expect(rule_effect).to have_received(:prepare).with(nil, workflow.id, subject.id).once
```

Comment:
Instead of expecting `nil` here, you could also pass in `id: 123` when we call `build`. But I don't mind too much. LGTM either way.

Reply:
that's much cleaner, you're right. i'm pushing a fix to that then i'm gonna merge